### ISSUE TITLE
ESP8266 1.10 polling causing crash

### ISSUE THREAD

**Issue by astrilchuk** [micropython/micropython#4511]
The following code works on my ESP8266 with 1.9.3 and 1.9.4 (using Wemos D1 Mini) but crashes using 1.10 on NodeMCU.  The reset is caused when calling `res = poll.poll(10000)`:

```
import usocket as socket
import uselect as select

addr = socket.getaddrinfo('0.0.0.0', 8000)[0][-1]
s = socket.socket()
s.setblocking(False)
s.bind(addr)
s.listen(1)

poll = select.poll()
poll.register(s, select.POLLIN)

res = poll.poll(10000)
```

Causes the following reset:

```
 ets Jan  8 2013,rst cause:2, boot mode:(3,6)

load 0x40100000, len 31020, room 16
tail 12
chksum 0xd2
ho 0 tail 12 room 4
load 0x3ffe8000, len 1100, room 12
tail 0
chksum 0x9a
load 0x3ffe8450, len 824, room 8
tail 0
chksum 0xbd
csum 0xbd
```

**Comment by mcauser** [comment]
Duplicate of #4508

**Comment by astrilchuk** [comment]
Sorry about the duplicates - I'm not sure what happened :-(

**Comment by dpgeorge** [comment]
Thanks for the report.  It's quite a bad bug, introduced in 29da9f06709df77d15437b67db5978ecafcccaef

It should now be fixed by 42c0e440b9635fb33c0b2a162f5bfe3122dce079

To workaround the error please use a backlog of 2 or more, eg: `s.listen(2)`.

**Comment by dpgeorge** [comment]
> To workaround the error please use a backlog of 2 or more

By that I mean that if you can't upgrade to the latest firmware (which fixes the bug) and need to use v1.10, then use a backlog of 2 or more.
